applications/ShallowWaterApplication/custom_utilities/post_process_utilities.h

```diff
@@ -0,0 +1,311 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Miguel Maso Sotomayor
+//
+
+#ifndef KRATOS_POST_PROCESS_UTILITIES_H_INCLUDED
+#define KRATOS_POST_PROCESS_UTILITIES_H_INCLUDED
+
+
+// System includes
+
+
+// External includes
+
+
+// Project includes
+#include "includes/define.h"
+#include "shallow_water_application_variables.h"
+
+
+namespace Kratos
+{
+///@addtogroup ShallowWaterApplication
+///@{
+
+///@name Kratos Globals
+///@{
+
+///@}
+///@name Type Definitions
+///@{
+
+///@}
+///@name  Enum's
+///@{
+
+///@}
+///@name  Functions
+///@{
+
+///@}
+///@name Kratos Classes
+///@{
+
+/// Auxiliary utilities for post process purpose.
+/** Gid is able to representate different entities when they have diffrent properties.
+ *  For each existing property, an auxiliary property is created, which means dry state.
+ *  Gid will print the wet domain (original property) with a property-color, and the dry
+ *  domain with another color.
+ */
+class PostProcessUtilities
+{
+public:
+    ///@name Type Definitions
+    ///@{
+
+    typedef std::size_t IndexType;
+
+    /// Pointer definition of PostProcessUtilities
+    KRATOS_CLASS_POINTER_DEFINITION(PostProcessUtilities);
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /// Default constructor.
+    PostProcessUtilities(ModelPart& rThisModelPart) : mrModelPart(rThisModelPart) {}
+
+    /// Destructor.
+    ~PostProcessUtilities() = default;
+
+    ///@}
+    ///@name Operators
+    ///@{
+
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    void DefineAuxiliaryProperties()
+    {
+        mWetToDryPropertiesMap.clear();
+        mDryToWetPropertiesMap.clear();
+
+        IndexType max_id = 0;
+        IndexVectorType prop_ids;
+
+        // Loop the original ids
+        for (IndexType i = 0; i < mrModelPart.NumberOfProperties(); ++i)
+        {
+            auto prop_id = (mrModelPart.PropertiesBegin() + i)->Id();
+            max_id = std::max(max_id, prop_id);
+            prop_ids.push_back(prop_id);
+        }
+
+        // Creating the auxiliary ids for the dry domain
+        for (auto id : prop_ids)
+        {
+            // Get pointers to the properties and create the dry property
+            Properties::Pointer wet_prop = mrModelPart.pGetProperties(id);
+            Properties::Pointer dry_prop(new Properties(*wet_prop));
```

Comment:
```suggestion
            auto dry_prop = Kratos::make_shared<Properties>(*wet_prop);
```
The shared pointers should now be constructed using the `Kratos::make_shared<Type>()` method.

Reply:
Fixed